### ISSUE TITLE
Add support for R in data viewer

### ISSUE THREAD

**Issue by gvelasq** [microsoft/vscode-jupyter#4604]
Currently when I run the Jupyter extension linked to the R kernel (IRkernel) in VS Code Insiders, I get the following error:

```{r}
View(mtcars)
ERROR: Error in View(mtcars): 'View()' not yet supported in the Jupyter R kernel
```

Adding data viewer support for the R kernel would allow the [REditorSupport](https://github.com/REditorSupport) developers to adapt the Jupyter data viewer for use with the VS Code R [extension](https://github.com/Ikuyadeu/vscode-R) and [debugger](https://github.com/ManuelHentschel/VSCode-R-Debugger). Hopefully the implementation would be similar to what has been done in the Python extension.

Related discussions:
1. In `vscode-jupyter`: https://github.com/microsoft/vscode-jupyter/issues/1536
2. In `vscode-R`: https://github.com/Ikuyadeu/vscode-R/issues/506
3. In `VSCode-R-Debugger`: https://github.com/ManuelHentschel/VSCode-R-Debugger/issues/123

**Comment by IanMatthewHuff** [comment]
@gvelasq Thanks for the info here. So that error is actually coming from the installed R Jupyter kernel I believe. In VS Code we're just launching whatever kernel is installed. If that kernel doesn't support View() then there is nothing that we could currently do for that. Looks like this issue here? But I might not be fully understanding what you are asking.
https://github.com/IRkernel/IRkernel/issues/280

I will say that we are looking at making our existing data and variable viewer support non-Python languages better. Due to the kernel issue with View() I don't think that we can use the same approach for data viewing support.

**Comment by IanMatthewHuff** [comment]
Yeah, we are very much looking at the aspects of our tool that currently only work with python (debugging, variable view, data viewer) and looking at how to expand those out to the entire ecosystem of languages. 

This probably isn't exactly what you need. But we do actually have an extensibility point for another VSCode extension to utilize our data viewer UI. You'd have to do most of the work yourself for providing the data but you could get the look and feel of our data viewer UI with this. 

https://github.com/microsoft/vscode-jupyter/blob/b368ea9326c13d6497e0f070d89d810d4ff86416/src/client/api.ts#L45
https://github.com/microsoft/vscode-jupyter/blob/b368ea9326c13d6497e0f070d89d810d4ff86416/src/client/datascience/data-viewing/types.ts#L59

We just did this to support a sister team, so we don't guarantee support for this, but might be helpful if you or anyone from the R world is looking to implement a stopgap solution until our full support comes online.

**Comment by RandomFractals** [comment]
looks like a neat interface. feel free to borrow and integrate some of the [data providers](https://github.com/RandomFractals/vscode-data-preview/tree/master/src/data.providers) I created for my generic [DataPreview](https://marketplace.visualstudio.com/items?itemName=RandomFractalsInc.vscode-data-preview) extension. should be an easy refactor to fit your interfaces.

Also, I might be switching them to [loaders.gl](https://github.com/visgl/loaders.gl/tree/master/modules) eventually. Check it out too if you want to cover most of the tabular and geo data formats.

**Comment by danielbasso** [comment]
Hi @IanMatthewHuff,

The interfaces exposed by `types.ts` need to be recreate inside another extension project or they can be referenced directly? Apparently they need to be recreated, but if you could clarify I appreciate.

**Comment by IanMatthewHuff** [comment]
@danielbasso Yeah, just recreating should be fine.

**Comment by gvelasq** [comment]
By way of follow up, support for `View()` was added in the development version of IRkernel.

Screenshots: https://github.com/IRkernel/IRkernel/issues/280#issuecomment-776057843

Commit: https://github.com/IRkernel/IRkernel/commit/8ac07057fdf134e24183b822d13bcbe3091bf76e

**Comment by DonJayamanne** [comment]
It has been over a year and there haven't been enough interest
Closing this for now.
If still required, please feel free to ping here and we'll re-open this issue

**Comment by gvelasq** [comment]
`View()` was officially implemented in IRkernel release [1.2](https://github.com/IRkernel/IRkernel/releases/tag/1.2). I haven't tested how it works on VS Code Insiders.
